workspace-manager-clienttests/src/main/java/scripts/uploadscripts/SummariesToBigQuery.java

```diff
@@ -0,0 +1,215 @@
+package scripts.uploadscripts;
+
+import bio.terra.testrunner.collector.MeasurementCollectionScript;
+import bio.terra.testrunner.collector.MeasurementCollector;
+import bio.terra.testrunner.common.utils.BigQueryUtils;
+import bio.terra.testrunner.runner.TestRunner;
+import bio.terra.testrunner.runner.TestScriptResult;
+import bio.terra.testrunner.runner.config.ServiceAccountSpecification;
+import bio.terra.testrunner.runner.config.TestConfiguration;
+import bio.terra.testrunner.runner.config.TestScriptSpecification;
+import bio.terra.testrunner.uploader.UploadScript;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.InsertAllRequest;
+import com.google.cloud.bigquery.TableId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.file.Path;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+
+public class SummariesToBigQuery extends UploadScript {
+    private static final Logger logger = LoggerFactory.getLogger(SummariesToBigQuery.class);
+
+    /** Public constructor so that this class can be instantiated via reflection. */
+    public SummariesToBigQuery() {}
+
+    protected String projectId; // google project id
+    protected String datasetName; // big query dataset name
+
+    protected TestConfiguration renderedTestConfiguration;
+    protected TestRunner.TestRunSummary testRunSummary;
+    protected MeasurementCollectionScript.MeasurementResultSummary[] measurementCollectionSummaries;
+
+    /**
+     * Setter for any parameters required by the upload script. These parameters will be set by the
+     * Result Uploader based on the current Upload List, and can be used by the upload script methods.
+     *
+     * @param parameters list of string parameters supplied by the upload list
+     */
+    public void setParameters(List<String> parameters) throws Exception {
+        if (parameters == null || parameters.size() < 2) {
+            throw new IllegalArgumentException(
+                    "Must provide BigQuery project_id and dataset_name in the parameters list");
+        }
+        projectId = parameters.get(0);
+        datasetName = parameters.get(1);
```

Comment:
is it possible to make these named arguments, not positional arguments? Is this a limitation of TestRunner itself?

Reply:
I have recorded this suggestion this [doc](https://docs.google.com/document/d/1gYqlJTpC4C4S1K0pYikeNJx-HWNZAZJYybinmuqcSTI/edit?usp=sharing) as a test runner enhancement.